w5e4.py

```diff
@@ -0,0 +1,102 @@
+# "w5e4"
+
+# connected vessels
+'''
+ Write a function called interleave that gets one or more iterable parameters, and returns a list of intertwined members.
+'''
+
+# function to interleave lists
+def interleave(*iterable):
+    longest = max(len(i) for i in iterable)
+    result = []
+    j = 0
+    #Run like this on the loop because it is important that the first members of each list appear first
+    while j < longest:
+        for i in range(len(iterable)):
+            if j < len(iterable[i]):
+                result.append(iterable[i][j])
+        j += 1
+
+    return result
+
+# print(interleave('abc', [1, 2, 3], ('!', '@', '#', '$')))
+
+# function to interleave lists
+def interleave_generator(*iterable):
+    longest = max(len(i) for i in iterable)
+    result = []
+    j = 0
+    while j < longest:
+        for i in range(len(iterable)):
+            if j < len(iterable[i]):
+                yield (iterable[i][j])
+        j += 1
+
+
+# our_generator = interleave_generator('abc', [1, 2, 3], ('!', '@', '#', '$'))
+# for i in our_generator:
+#     print(i)
+
+
+#Harry is rational but not terrible
+"""
+The librarian who was on his way to bring me the coveted copy of "Harry Potter and the Rational Method" stumbled.
```

Comment:
Im really hoping you didnt translate all of that on your own :) You dont have to do that